synapse/rest/client/v2_alpha/report_event.py

```diff
@@ -31,16 +44,39 @@
 class ReportEventRestServlet(RestServlet):
     PATTERNS = client_patterns("/rooms/(?P<room_id>[^/]*)/report/(?P<event_id>[^/]*)$")
 
+    # As we cannot import HomeServer without creating a circular dependency,
+    # use a forward reference.
+    hs: 'HomeServer'
+    auth: Auth
+    federation_sender: Optional[FederationSender]
+    server_notices: ServerNoticesManager
+    state: StateHandler
+    storage: Storage
+    store: DataStore
+
     def __init__(self, hs):
         super().__init__()
         self.hs = hs
         self.auth = hs.get_auth()
+
+        self.federation_sender = None
+        if hs.should_send_federation():
+            self.federation_sender = hs.get_federation_sender()
+            hs.get_federation_registry().register_edu_handler(
+                "org.matrix.m.content_report", self.on_receive_report_through_federation
+            )
```

Comment:
note that whether you will *receive* EDUs on this worker is unrelated to `hs.should_send_federation`:

```suggestion
        hs.get_federation_registry().register_edu_handler(
            "org.matrix.m.content_report", self.on_receive_report_through_federation
        )
```